### ISSUE TITLE
Segfault when using smtp_server

### ISSUE THREAD

**Issue by WilliamDEdwards** [acassen/keepalived#1530]
**Describe the bug**

When `smtp_server` is defined, keepalived on Debian 10 segfaults on startup.

The same configuration is used on Debian 9, with no issues.

**To Reproduce**

Specify the `smtp_server` directive with a value (either a hostname, IPv4 or IPv6 address) in the keepalived configuration file.

**Expected behavior**

I would expect keepalived to start with this configuration, as the same configuration is functional on Debian 9.

**Keepalived version**

```
root@http01:/etc/keepalived# keepalived -v
Keepalived v2.0.10 (11/12,2018)

Copyright(C) 2001-2018 Alexandre Cassen, <acassen@gmail.com>

Built with kernel headers for Linux 4.18.20
Running on Linux 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26)

configure options: --build=x86_64-linux-gnu --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-silent-rules --libdir=${prefix}/lib/x86_64-linux-gnu --libexecdir=${prefix}/lib/x86_64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --with-kernel-dir=debian/ --enable-snmp --enable-sha1 --enable-snmp-rfcv2 --enable-snmp-rfcv3 --enable-dbus --enable-dbus-create-instance --enable-json --enable-bfd build_alias=x86_64-linux-gnu CFLAGS=-g -O2 -fdebug-prefix-map=/build/keepalived-8agDac/keepalived-2.0.10=. -fstack-protector-strong -Wformat -Werror=format-security LDFLAGS=-Wl,-z,relro CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2

Config options:  LIBIPSET_DYNAMIC LVS VRRP VRRP_AUTH JSON BFD OLD_CHKSUM_COMPAT FIB_ROUTING SNMP_V3_FOR_V2 SNMP_VRRP SNMP_CHECKER SNMP_RFCV2 SNMP_RFCV3 DBUS DBUS_CREATE_INSTANCE

System options:  PIPE2 SIGNALFD INOTIFY_INIT1 VSYSLOG EPOLL_CREATE1 IPV4_DEVCONF LIBNL3 RTA_ENCAP RTA_EXPIRES RTA_NEWDST RTA_PREF FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP FRA_TUN_ID RTAX_CC_ALGO RTAX_QUICKACK RTEXT_FILTER_SKIP_STATS FRA_L3MDEV FRA_UID_RANGE RTAX_FASTOPEN_NO_COOKIE RTA_VIA FRA_OIFNAME FRA_PROTOCOL FRA_IP_PROTO FRA_SPORT_RANGE FRA_DPORT_RANGE RTA_TTL_PROPAGATE IFA_FLAGS IP_MULTICAST_ALL LWTUNNEL_ENCAP_MPLS LWTUNNEL_ENCAP_ILA LIBIPTC LIBIPSET_PRE_V7 LIBIPVS_NETLINK IPVS_DEST_ATTR_ADDR_FAMILY IPVS_SYNCD_ATTRIBUTES IPVS_64BIT_STATS VRRP_VMAC SOCK_NONBLOCK SOCK_CLOEXEC O_PATH GLOB_BRACE INET6_ADDR_GEN_MODE VRF SO_MARK SCHED_RT SCHED_RESET_ON_FORK
```

**Distro (please complete the following information):**

 - Name: Debian
 - Version:

```
root@http01:/etc/keepalived# uname -r
4.19.0-8-amd64
root@http01:/etc/keepalived# lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
```

 - Architecture: x86_64

**Details of any containerisation or hosted service (e.g. AWS)**

Of course not. :)

**Configuration file:**

```
global_defs {
    notification_email {
        engineering@cyberfusion.nl
    }
    notification_email_from engineering@cyberfusion.nl
    smtp_server smtp.prorelay.nl 587
    smtp_helo_name http01.panel.cyberfusion.space
    smtp_connect_timeout 30
    router_id http01.panel.cyberfusion.space
    enable_traps
    enable_script_security
    vrrp_version 3
    vrrp_strict
    vrrp_iptables
    vrrp_ipsets
}

vrrp_script chk {
    script "/usr/bin/pkill -0 haproxy"
    interval 2
}

vrrp_instance panel-vip_4 {
    state MASTER
    interface ens18
    virtual_router_id 125
    accept
    smtp_alert

    virtual_ipaddress {
        185.233.175.27/32
    }
    track_script {
        chk
    }
}

vrrp_instance panel-vip_6 {
    state MASTER
    interface ens18
    virtual_router_id 125
    accept
    smtp_alert

    virtual_ipaddress {
        2a0c:eb00:0:f6:185:233:175:27/128
    }
    track_script {
        chk
    }
}
```

**Notify and track scripts**

`pkill`

**System Log entries**

```
Apr  2 10:50:17 http01 Keepalived[1576]: Starting Keepalived v2.0.10 (11/12,2018)
Apr  2 10:50:17 http01 Keepalived[1576]: Running on Linux 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) (built for Linux 4.18.20)
Apr  2 10:50:17 http01 Keepalived[1576]: Command line: 'keepalived'
Apr  2 10:50:17 http01 Keepalived[1576]: Opening file '/etc/keepalived/keepalived.conf'.
Apr  2 10:50:17 http01 kernel: [  470.442450] keepalived[1576]: segfault at 0 ip 00007f55e0222864 sp 00007ffea7d19c28 error 4 in libc-2.28.so[7f55e01b5000+148000]
Apr  2 10:50:17 http01 kernel: [  470.442457] Code: 00 e9 4a ff ff ff 66 0f 6f 0c 0e 66 0f 73 d8 01 66 0f 73 db 01 e9 7c 11 00 00 66 66 2e 0f 1f 84 00 00 00 00 00 90 66 0f ef c0 <66> 0f 6f 17 66 0f 6f 0e 66 0f 74 c1 66 0f 73 fa 0e 66 0f 74 d1 66
```

**Additional context**

End of strace when `smtp_server` is set:

```
futex(0x7f4677f82044, FUTEX_WAKE_PRIVATE, 2147483647) = 0
socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC, IPPROTO_IP) = 6
connect(6, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("185.233.175.192")}, 16) = 0
getsockname(6, {sa_family=AF_INET, sin_port=htons(56900), sin_addr=inet_addr("185.233.175.26")}, [28->16]) = 0
close(6)                                = 0
socket(AF_INET6, SOCK_DGRAM|SOCK_CLOEXEC, IPPROTO_IP) = 6
connect(6, {sa_family=AF_INET6, sin6_port=htons(0), inet_pton(AF_INET6, "2a0c:eb00:0:f7:185:233:175:192", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
getsockname(6, {sa_family=AF_INET6, sin6_port=htons(40322), inet_pton(AF_INET6, "2a0c:eb00:0:f6:185:233:175:26", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, [28]) = 0
close(6)                                = 0
read(4, "", 4096)                       = 0
close(4)                                = 0
fchdir(5)                               = 0
close(5)                                = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=NULL} ---
+++ killed by SIGSEGV +++
Segmentation fault
```

End of strace when `smtp_server` is not set:

```
connect(3, {sa_family=AF_UNIX, sun_path="/dev/log"}, 110) = 0
sendto(3, "<30>Apr  2 10:51:35 Keepalived[1"..., 78, MSG_NOSIGNAL, NULL, 0) = 78
getpid()                                = 1677
sendto(3, "<30>Apr  2 10:51:35 Keepalived[1"..., 132, MSG_NOSIGNAL, NULL, 0) = 132
getpid()                                = 1677
sendto(3, "<30>Apr  2 10:51:35 Keepalived[1"..., 64, MSG_NOSIGNAL, NULL, 0) = 64
lstat("/etc/keepalived/keepalived.conf", {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
stat("/etc/keepalived/keepalived.conf", {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
access("/etc/keepalived/keepalived.conf", R_OK) = 0
stat("/etc/keepalived/keepalived.conf", {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
lstat("/etc/keepalived/keepalived.conf", {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
stat("/etc/keepalived/keepalived.conf", {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
getpid()                                = 1677
sendto(3, "<30>Apr  2 10:51:35 Keepalived[1"..., 85, MSG_NOSIGNAL, NULL, 0) = 85
openat(AT_FDCWD, "/etc/keepalived/keepalived.conf", O_RDONLY) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
openat(AT_FDCWD, ".", O_RDONLY|O_PATH|O_DIRECTORY) = 5
chdir("/etc/keepalived")                = 0
fstat(4, {st_mode=S_IFREG|0644, st_size=932, ...}) = 0
read(4, "global_defs {\n    notification_e"..., 4096) = 932
read(4, "", 4096)                       = 0
close(4)                                = 0
fchdir(5)                               = 0
close(5)                                = 0
openat(AT_FDCWD, "/var/run/keepalived.pid", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/run/vrrp.pid", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/run/checkers.pid", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/run/bfd.pid", O_RDONLY) = -1 ENOENT (No such file or directory)
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f1a5fde8290) = 1678
close(3)                                = 0
close(0)                                = 0
close(1)                                = 0
close(2)                                = 0
exit_group(0)                           = ?
+++ exited with 0 +++
```

**Comment by pqarmitage** [comment]
This issue was resolved by commit fccdeba which was first included in keepalvied v2.0.11.

Could you please file this as a bug against Debian keepalived and request that they backport commit fccdeba.

**Comment by WilliamDEdwards** [comment]
> This issue was resolved by commit [fccdeba](https://github.com/acassen/keepalived/commit/fccdeba30c58927da00f2d3849155341298c7a7f) which was first included in keepalvied v2.0.11.

Ah, I see. I did not see that commit, despite my searches :)

> 
> Could you please file this as a bug against Debian keepalived and request that they backport commit [fccdeba](https://github.com/acassen/keepalived/commit/fccdeba30c58927da00f2d3849155341298c7a7f).

Done: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=958898
